src-nodejs/flureenjs.cljs

```diff
@@ -81,11 +81,26 @@
 
 ;; define your app data so that it doesn't get over-written on reload
 (defonce app-state (atom {:product "Fluree NodeJs Library"
-                          :version "v1.0.0-rc5"}))
+                          :version "v1.0.0-rc16"}))
 
 (println (:product @app-state) (:version @app-state))
 
 
+;; ======================================
+;;
+;; Get handle to Node.js crypto module
+;;
+;; ======================================
+
+(defonce njs-crypto (atom {:crypto nil}))
+(try
+  (let [cm (js/require "crypto")]
+    (swap! njs-crypto assoc :crypto cm))
+  (catch :default ex
+    (log/warn (str "Error: Unable to access Node.js crypto module:" ex))
+    (log/warn "Private key generation is not available.")))
```

Comment:
Not a blocker, but I think you should return 'nil' in the exception case. Because we are not throwing here, your var will get whatever the last form executed is... in this case it will get whatever (log/warn... ) returns. While I believe that is 'nil' today, no guarantee someone won't change that in the future.

Reply:
Sorry Brian, 
I missed this comment.  The code looks a little different now.  I do set njs-crypto to nil:

```
(def ^:const njs-crypto
  (try
    (js/require "crypto")
    (catch :default ex
      (do
        (log/warn (str "Error: Unable to access Node.js crypto module:" ex))
        (log/warn "Private key generation is not available.")
        nil))))
```
      
      And then, an error will be thrown if a key pair is requested:
      
 ```
(defn- generate-key-pair
  "Generates a private-public key pair using the Node.js
  crypto module. The JavaScript code looks like:

     const ecdh = crypto.createECDH('secp256k1');
     ecdh.generateKeys()
     return { private: ecdh.getPrivateKey('hex'),
              public:  ecdh.getPublicKey('hex','compressed')};
  "
  []
  (if njs-crypto
    (let [ecdh      (js-invoke njs-crypto "createECDH" "secp256k1")
          _         (js-invoke ecdh "generateKeys")]
      {:private (js-invoke ecdh "getPrivateKey" "hex")
       :public (js-invoke ecdh "getPublicKey" "hex" "compressed")})
    (throw "Node.js crypto module not accessible")))
```